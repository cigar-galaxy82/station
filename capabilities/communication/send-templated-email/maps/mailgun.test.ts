import { sendTemplatedEmailTest } from './send-templated-email';

sendTemplatedEmailTest(
  'mailgun',
  {
    from: 'demo@demo.superface.org',
    to: 'demo@superface.ai',
    templateId: 'station-test',
  },
  {
    beforeRecordingSave: recordings => {
      recordings.forEach(recording => {
        recording.path = recording.path.replace(
          process.env.MAILGUN_DOMAIN as string,
          'example.com'
        );
      });
    },
    afterRecordingLoad: scopes => {
      scopes.forEach(scope => {
        scope.filteringPath(
          new RegExp(process.env.MAILGUN_DOMAIN as string, 'g'),
          'example.com'
        );
      });
    },
  }
);
