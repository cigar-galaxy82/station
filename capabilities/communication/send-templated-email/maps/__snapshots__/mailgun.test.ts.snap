// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`communication/send-templated-email/mailgun SendTemplatedEmail when all inputs are correct should return messageId as result 1`] = `
Object {
  "value": Object {
    "messageId": "<20211103112326.1.701865F1F9FD7C61@demo.superface.org>",
  },
}
`;

exports[`communication/send-templated-email/mailgun SendTemplatedEmail when inputs are invalid should throw error 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
Properites: {
  \\"title\\": \\"Invalid inputs\\",
  \\"detail\\": \\"'from' parameter is not a valid address. please check documentation\\"
}
AST Path: definitions[0].statements[0].responseHandlers[1].statements[0]
Original Map Location: Line 27, column 7",
}
`;
